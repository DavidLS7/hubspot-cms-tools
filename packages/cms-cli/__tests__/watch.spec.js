/*global banjo */

describe('Watch command', () => {
  it('works with help positional', async () =>
    expect(await banjo(['watch', 'help'])).toMatchSnapshot());

  it('works with help option', async () =>
    expect(await banjo(['watch', '--help'])).toMatchSnapshot());
});
