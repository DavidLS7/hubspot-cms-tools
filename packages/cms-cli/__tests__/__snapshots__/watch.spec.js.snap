// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Watch command works with help option 1`] = `
"banjo watch <src> <dest>

Watch a directory on your computer for changes and upload the changed files to
the HubSpot CMS

Positionals:
  src   Path to the local directory your files are in, relative to your current
        working directory                                    [string] [required]
  dest  Path in HubSpot Design Tools. Can be a net new path  [string] [required]

Options:
  --version             Show version number                            [boolean]
  --debug, -d           set log level to debug        [boolean] [default: false]
  --help                Show help                                      [boolean]
  --config, -c          path to a config file                           [string]
  --portal, -p          HubSpot portal id or name from config           [string]
  --mode, -m            write to <draft | publish>                      [string]
  --remove, -r          Will cause watch to delete files in your HubSpot account
                        that are not found locally.                    [boolean]
  --initial-upload, -i  Upload directory before watching for updates   [boolean]
  --notify, -n          Log to specified file when a watch task is triggered and
                        after workers have gone idle. Ex. --notify path/to/file
                                                                        [string]
"
`;

exports[`Watch command works with help positional 1`] = `
"banjo watch <src> <dest>

Watch a directory on your computer for changes and upload the changed files to
the HubSpot CMS

Positionals:
  src   Path to the local directory your files are in, relative to your current
        working directory                                    [string] [required]
  dest  Path in HubSpot Design Tools. Can be a net new path  [string] [required]

Options:
  --version             Show version number                            [boolean]
  --debug, -d           set log level to debug        [boolean] [default: false]
  --help                Show help                                      [boolean]
  --config, -c          path to a config file                           [string]
  --portal, -p          HubSpot portal id or name from config           [string]
  --mode, -m            write to <draft | publish>                      [string]
  --remove, -r          Will cause watch to delete files in your HubSpot account
                        that are not found locally.                    [boolean]
  --initial-upload, -i  Upload directory before watching for updates   [boolean]
  --notify, -n          Log to specified file when a watch task is triggered and
                        after workers have gone idle. Ex. --notify path/to/file
                                                                        [string]
"
`;
